main.go

```diff
@@ -1,133 +1,151 @@
-package main
-
-import (
-	"encoding/json"
-	"fmt"
-	"io/ioutil"
-	"log"
-	"net/http"
-	"sync"
-	"strconv"
-
-	"github.com/gorilla/mux"
-)
-
-type autoInc struct {
-    sync.Mutex

```

Comment:
This is clever. I feel like I always make a real mutex. But its so unnecessary. This is slick.